lib/Subscribers/SubscribersRepository.php

```diff
@@ -112,13 +113,14 @@ public function bulkDelete(array $ids): int {
       ", ['ids' => $ids], ['ids' => Connection::PARAM_INT_ARRAY]);
 
       $queryBuilder = $entityManager->createQueryBuilder();
-      $queryBuilder->delete(SubscriberEntity::class, 's')
+      $count = $queryBuilder->delete(SubscriberEntity::class, 's')
         ->where('s.id IN (:ids)')
+        ->andWhere('s.wpUserId IS NULL')
```

Comment:
I think this is something that should be covered with a test. We have [a test for this repository](https://github.com/mailpoet/mailpoet/blob/master/tests/integration/Subscribers/SubscribersRepositoryTest.php). What do you think about extending it with this use-case?

Reply:
Good idea, I'll add two or three tests for this use case.